java.lang.NumberFormatException: For input string: "8849485702" at InstanceId.<init>(InstanceId.java:65)

## Bug Report

This issue affects ShardingSphere >= 5.1.0

Caused by: https://github.com/apache/shardingsphere/pull/15188

Relates to #17806.

### Which version of ShardingSphere did you use?

master - https://github.com/apache/shardingsphere/tree/a393962f5b3eec2d154ad1c2c5d15a758efc499b

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Scaling (ShardingSphere-JDBC is used in Scaling)

### Actual behavior

After data consistency check kept failing for 12 hours, new error occurred.

```
[INFO ] 2022-05-20 09:51:00.032 [_finished_check_Worker-1] o.a.s.d.p.core.job.FinishedCheckJob - scaling job 0130317c30317c3054317c626d73716c5f7368617264696e67 almost finished.
[INFO ] 2022-05-20 09:51:00.032 [_finished_check_Worker-1] o.a.s.d.p.s.r.s.DefaultSourceWritingStopLock - lock, databaseName=bmsql_sharding, jobId=0130317c30317c3054317c626d73716c5f7368617264696e67
[INFO ] 2022-05-20 09:51:00.054 [_finished_check_Worker-1] o.a.s.d.p.c.a.i.RuleAlteredJobAPIImpl - stopClusterWriteDB, tryLockSuccess=true
[INFO ] 2022-05-20 09:51:00.056 [_finished_check_Worker-1] o.a.s.d.p.core.job.FinishedCheckJob - dataConsistencyCheck for job 0130317c30317c3054317c626d73716c5f7368617264696e67
[INFO ] 2022-05-20 09:51:00.057 [_finished_check_Worker-1] o.a.s.d.p.a.c.TableNameSchemaNameMapping - mapping={pg_trigger=pg_catalog, bmsql_district=public, bmsql_oorder=public, bmsql_stock=public, columns=information_schema, pg_class=pg_catalog, pg_tablespace=pg_catalog, pg_database=pg_catalog, bmsql_customer=public, bmsql_item=public, tables=information_schema, bmsql_new_order=public, bmsql_history=public, bmsql_warehouse=public, pg_inherits=pg_catalog, bmsql_order_line=public, views=information_schema, bmsql_config=public}
[INFO ] 2022-05-20 09:51:00.060 [_finished_check_Worker-1] o.a.s.d.p.s.r.s.DefaultSourceWritingStopLock - releaseLock, databaseName=bmsql_sharding, jobId=0130317c30317c3054317c626d73716c5f7368617264696e67
[INFO ] 2022-05-20 09:51:00.060 [_finished_check_Worker-1] o.a.s.d.p.c.a.i.RuleAlteredJobAPIImpl - restoreClusterWriteDB, before releaseLock, databaseName=bmsql_sharding, jobId=0130317c30317c3054317c626d73716c5f7368617264696e67
[ERROR] 2022-05-20 09:51:00.062 [_finished_check_Worker-1] o.a.s.d.p.core.job.FinishedCheckJob - scaling job 0130317c30317c3054317c626d73716c5f7368617264696e67 finish check failed!
java.lang.NumberFormatException: For input string: "8849485702"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Integer.parseInt(Integer.java:668)
	at java.base/java.lang.Integer.valueOf(Integer.java:999)
	at org.apache.shardingsphere.infra.instance.definition.InstanceId.<init>(InstanceId.java:65)
	at org.apache.shardingsphere.infra.instance.definition.InstanceDefinition.<init>(InstanceDefinition.java:36)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.createContextManager(ShardingSphereDataSource.java:78)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.<init>(ShardingSphereDataSource.java:64)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:77)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:138)
	at org.apache.shardingsphere.driver.data.pipeline.datasource.creator.ShardingSpherePipelineDataSourceCreator.createPipelineDataSource(ShardingSpherePipelineDataSourceCreator.java:43)
	at org.apache.shardingsphere.data.pipeline.core.datasource.PipelineDataSourceFactory.newInstance(PipelineDataSourceFactory.java:44)
	at org.apache.shardingsphere.data.pipeline.core.check.consistency.DataConsistencyChecker.checkCount(DataConsistencyChecker.java:111)
	at org.apache.shardingsphere.data.pipeline.core.check.consistency.DataConsistencyChecker.check(DataConsistencyChecker.java:93)
	at org.apache.shardingsphere.data.pipeline.core.api.impl.RuleAlteredJobAPIImpl.dataConsistencyCheck(RuleAlteredJobAPIImpl.java:288)
	at org.apache.shardingsphere.data.pipeline.core.api.impl.RuleAlteredJobAPIImpl.dataConsistencyCheck(RuleAlteredJobAPIImpl.java:274)
	at org.apache.shardingsphere.data.pipeline.core.job.FinishedCheckJob.dataConsistencyCheck(FinishedCheckJob.java:124)
	at org.apache.shardingsphere.data.pipeline.core.job.FinishedCheckJob.execute(FinishedCheckJob.java:89)
	at org.apache.shardingsphere.elasticjob.simple.executor.SimpleJobExecutor.process(SimpleJobExecutor.java:33)
	at org.apache.shardingsphere.elasticjob.simple.executor.SimpleJobExecutor.process(SimpleJobExecutor.java:29)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.process(ElasticJobExecutor.java:172)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.process(ElasticJobExecutor.java:141)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.execute(ElasticJobExecutor.java:124)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.execute(ElasticJobExecutor.java:100)
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJob.execute(LiteJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
```

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/instance/definition/InstanceDefinition.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/instance/definition/InstanceDefinition.java
@@ -38,7 +38,7 @@ public final class InstanceDefinition {
     
     public InstanceDefinition(final InstanceType instanceType, final Integer port) {
         this.instanceType = instanceType;
-        instanceId = new InstanceId(port);
+        instanceId = new InstanceId(String.valueOf(port));
     }
     
     public InstanceDefinition(final InstanceType instanceType, final String id) {

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/instance/definition/InstanceId.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/instance/definition/InstanceId.java
@@ -44,26 +44,25 @@ public final class InstanceId {
     public InstanceId(final String ip, final String uniqueSign) {
         this.ip = ip;
         this.uniqueSign = uniqueSign;
-        this.id = String.join(this.DELIMITER, ip, uniqueSign);
+        id = String.join(DELIMITER, ip, uniqueSign);
     }
     
     public InstanceId(final String instance) {
         if (instance.indexOf(DELIMITER) >= 0) {
-            this.id = instance;
+            id = instance;
             List<String> ids = Splitter.on("@").splitToList(instance);
-            this.ip = ids.get(0);
-            this.uniqueSign = ids.get(1);
+            ip = ids.get(0);
+            uniqueSign = ids.get(1);
         } else {
-            this.uniqueSign = instance;
-            this.ip = IpUtils.getIp();
-            this.id = String.join(this.DELIMITER, this.ip, this.uniqueSign);
+            uniqueSign = instance;
+            ip = IpUtils.getIp();
+            id = String.join(DELIMITER, ip, uniqueSign);
         }
     }
     
     public InstanceId() {
-        this.ip = IpUtils.getIp();
-        this.uniqueSign = String.join("", ManagementFactory.getRuntimeMXBean().getName().split(this.DELIMITER)[0], 
-                String.valueOf(this.ATOMIC_LONG.incrementAndGet()));
-        this.id = String.join(this.DELIMITER, this.ip, this.uniqueSign);
+        ip = IpUtils.getIp();
+        uniqueSign = String.join("", ManagementFactory.getRuntimeMXBean().getName().split(DELIMITER)[0], String.valueOf(ATOMIC_LONG.incrementAndGet()));
+        id = String.join(DELIMITER, ip, uniqueSign);
     }
 }

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/instance/definition/InstanceId.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/instance/definition/InstanceId.java
@@ -39,30 +39,31 @@ public final class InstanceId {
     
     private final String ip;
     
-    private final Integer uniqueSign;
+    private final String uniqueSign;
     
-    public InstanceId(final String ip, final Integer uniqueSign) {
+    public InstanceId(final String ip, final String uniqueSign) {
         this.ip = ip;
         this.uniqueSign = uniqueSign;
-        id = String.join(DELIMITER, ip, String.valueOf(uniqueSign));
+        this.id = String.join(this.DELIMITER, ip, uniqueSign);
     }
     
-    public InstanceId(final Integer uniqueSign) {
-        this.uniqueSign = uniqueSign;
-        ip = IpUtils.getIp();
-        id = String.join(DELIMITER, ip, String.valueOf(uniqueSign));
-    }
-    
-    public InstanceId(final String id) {
-        this.id = id;
-        List<String> ids = Splitter.on("@").splitToList(id);
-        ip = ids.get(0);
-        uniqueSign = Integer.valueOf(ids.get(1));
+    public InstanceId(final String instance) {
+        if (instance.indexOf(DELIMITER) >= 0) {
+            this.id = instance;
+            List<String> ids = Splitter.on("@").splitToList(instance);
+            this.ip = ids.get(0);
+            this.uniqueSign = ids.get(1);
+        } else {
+            this.uniqueSign = instance;
+            this.ip = IpUtils.getIp();
+            this.id = String.join(this.DELIMITER, this.ip, this.uniqueSign);
+        }
     }
     
     public InstanceId() {
-        ip = IpUtils.getIp();
-        uniqueSign = Integer.valueOf(String.join("", ManagementFactory.getRuntimeMXBean().getName().split(DELIMITER)[0], String.valueOf(ATOMIC_LONG.incrementAndGet())));
-        id = String.join(DELIMITER, ip, String.valueOf(uniqueSign));
+        this.ip = IpUtils.getIp();
+        this.uniqueSign = String.join("", ManagementFactory.getRuntimeMXBean().getName().split(this.DELIMITER)[0], 
+                String.valueOf(this.ATOMIC_LONG.incrementAndGet()));
+        this.id = String.join(this.DELIMITER, this.ip, this.uniqueSign);
     }
 }

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/instance/definition/InstanceIdTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/instance/definition/InstanceIdTest.java
@@ -27,8 +27,8 @@ public final class InstanceIdTest {
     
     @Test
     public void assertInitInstanceIdWithPort() {
-        InstanceId actual = new InstanceId(3307);
-        assertThat(actual.getUniqueSign(), is(3307));
+        InstanceId actual = new InstanceId("3307");
+        assertThat(actual.getUniqueSign(), is("3307"));
     }
     
     @Test
@@ -39,7 +39,7 @@ public final class InstanceIdTest {
     
     @Test
     public void assertInitInstanceIdWithIpAndUniqueSign() {
-        InstanceId actual = new InstanceId("127.0.0.1", 3307);
+        InstanceId actual = new InstanceId("127.0.0.1", "3307");
         assertThat(actual.getId(), is("127.0.0.1@3307"));
     }
     
@@ -47,7 +47,7 @@ public final class InstanceIdTest {
     public void assertInitInstanceIdWithExistId() {
         InstanceId actual = new InstanceId("127.0.0.1@3307");
         assertThat(actual.getIp(), is("127.0.0.1"));
-        assertThat(actual.getUniqueSign(), is(3307));
+        assertThat(actual.getUniqueSign(), is("3307"));
     }
     
     @Test

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/status/compute/subscriber/ComputeNodeStatusSubscriber.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/status/compute/subscriber/ComputeNodeStatusSubscriber.java
@@ -49,7 +49,7 @@ public final class ComputeNodeStatusSubscriber {
      */
     @Subscribe
     public void update(final ComputeNodeStatusChangedEvent event) {
-        String computeStatusNodePath = ComputeNode.getInstanceStatusNodePath(new InstanceId(event.getIp(), Integer.valueOf(event.getPort())).getId());
+        String computeStatusNodePath = ComputeNode.getInstanceStatusNodePath(new InstanceId(event.getIp(), String.valueOf(event.getPort())).getId());
         String yamlContext = repository.get(computeStatusNodePath);
         Collection<String> status = Strings.isNullOrEmpty(yamlContext) ? new ArrayList<>() : YamlEngine.unmarshal(yamlContext, Collection.class);
         if (event.getStatus() == ComputeNodeStatus.CIRCUIT_BREAK) {

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/test/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/ClusterContextManagerCoordinatorTest.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/test/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/ClusterContextManagerCoordinatorTest.java
@@ -310,7 +310,7 @@ public final class ClusterContextManagerCoordinatorTest {
     @Test
     public void assertRenewInstanceOfflineEvent() {
         coordinator.renew(new InstanceOfflineEvent(contextManager.getInstanceContext().getInstance().getInstanceDefinition()));
-        assertThat(contextManager.getInstanceContext().getInstance().getInstanceDefinition().getInstanceId().getUniqueSign(), is(3307));
+        assertThat(contextManager.getInstanceContext().getInstance().getInstanceDefinition().getInstanceId().getUniqueSign(), is("3307"));
     }
     
     @Test

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/LabelInstanceHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/LabelInstanceHandler.java
@@ -42,7 +42,7 @@ public final class LabelInstanceHandler extends UpdatableRALBackendHandler<Label
         if (null == persistService || null == persistService.getRepository() || persistService.getRepository() instanceof StandalonePersistRepository) {
             throw new UnsupportedOperationException("Labels can only be added in cluster mode");
         }
-        String instanceId = new InstanceId(sqlStatement.getIp(), Integer.valueOf(sqlStatement.getPort())).getId();
+        String instanceId = new InstanceId(sqlStatement.getIp(), String.valueOf(sqlStatement.getPort())).getId();
         ComputeNodeInstance instances = persistService.getComputeNodePersistService().loadComputeNodeInstance(new InstanceDefinition(InstanceType.PROXY, instanceId));
         Collection<String> labels = new LinkedHashSet<>(sqlStatement.getLabels());
         if (!sqlStatement.isOverwrite()) {

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/SetInstanceStatusHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/SetInstanceStatusHandler.java
@@ -44,7 +44,7 @@ public final class SetInstanceStatusHandler extends UpdatableRALBackendHandler<S
         if (!"Cluster".equals(contextManager.getInstanceContext().getModeConfiguration().getType())) {
             throw new UnsupportedOperationException("Only allowed in cluster mode");
         }
-        InstanceId operationInstanceId = new InstanceId(sqlStatement.getIp(), Integer.valueOf(sqlStatement.getPort()));
+        InstanceId operationInstanceId = new InstanceId(sqlStatement.getIp(), String.valueOf(sqlStatement.getPort()));
         boolean isDisable = "DISABLE".equals(sqlStatement.getStatus());
         if (isDisable) {
             checkDisablingIsValid(operationInstanceId);

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/UnlabelInstanceHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/UnlabelInstanceHandler.java
@@ -43,7 +43,7 @@ public final class UnlabelInstanceHandler extends UpdatableRALBackendHandler<Unl
         if (null == persistService || null == persistService.getRepository() || persistService.getRepository() instanceof StandalonePersistRepository) {
             throw new UnsupportedOperationException("Labels can only be removed in cluster mode");
         }
-        String instanceId = new InstanceId(sqlStatement.getIp(), Integer.valueOf(sqlStatement.getPort())).getId();
+        String instanceId = new InstanceId(sqlStatement.getIp(), String.valueOf(sqlStatement.getPort())).getId();
         ComputeNodeInstance instances = persistService.getComputeNodePersistService().loadComputeNodeInstance(new InstanceDefinition(InstanceType.PROXY, instanceId));
         Collection<String> labels = new LinkedHashSet<>(instances.getLabels());
         if (sqlStatement.getLabels().isEmpty()) {

